src/transformers/models/superglue/modeling_superglue.py

```diff
@@ -0,0 +1,845 @@
+        matching_scores (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Scores of predicted matches.
+        keypoints (`torch.FloatTensor` of shape `(batch_size, num_keypoints, 2)`):
+            Absolute (x, y) coordinates of predicted keypoints in a given image.
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    mask: torch.FloatTensor = None
+    matches: torch.FloatTensor = None
+    matching_scores: torch.FloatTensor = None
+    keypoints: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+class SuperGlueMultiLayerPerceptron(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+        channels: List[int],
+        do_batch_norm: bool = True,
+    ):
+        super().__init__()
+        num_layers = len(channels)
+        layers = []
+        for i in range(1, num_layers):
+            layers.append(nn.Conv1d(channels[i - 1], channels[i], kernel_size=1, bias=True))
+            if i < (num_layers - 1):
+                if do_batch_norm:
+                    layers.append(nn.BatchNorm1d(channels[i]))
+                layers.append(nn.ReLU())
+        self.layers = nn.Sequential(*layers)
+        nn.init.constant_(self.layers[-1].bias, 0.0)
+
+    def forward(
+        self,
+        input: torch.Tensor,
+        output_hidden_states: Optional[bool] = False,
+    ) -> Union[Tuple, torch.Tensor]:
+        all_hidden_states = () if output_hidden_states else None
+        for layer in self.layers:
+            input = layer(input)
+            if output_hidden_states and isinstance(layer, nn.Conv1d):
+                all_hidden_states = all_hidden_states + (input,)
+        output = input
+        return output, all_hidden_states
+
+
+class SuperGlueKeypointEncoder(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+    ):
+        super().__init__()
+        self.layer_sizes = config.keypoint_encoder_sizes
+        self.feature_dim = config.descriptor_dim
+        self.encoder = SuperGlueMultiLayerPerceptron(config, [3] + self.layer_sizes + [self.feature_dim])
+
+    def forward(
+        self,
+        keypoints: torch.Tensor,
+        scores: torch.Tensor,
+        output_hidden_states: Optional[bool] = False,
+    ) -> torch.Tensor:
+        keypoints = keypoints.transpose(1, 2)
+        scores = scores.unsqueeze(1)
+        inputs = torch.cat([keypoints, scores], dim=1)
+        return self.encoder(inputs, output_hidden_states=output_hidden_states)
+
+
+class SuperGlueMultiHeadAttention(nn.Module):
+    def __init__(self, config: SuperGlueConfig):
+        super().__init__()
+        self.feature_dim = config.descriptor_dim
+        self.num_heads = config.num_heads
+        self.dim = self.feature_dim // self.num_heads
+        self.merge = nn.Conv1d(self.feature_dim, self.feature_dim, kernel_size=1)
```

Comment:
```suggestion
        feature_dim = config.descriptor_dim
        self.num_heads = config.num_heads
        self.dim = self.feature_dim // self.num_heads
        self.merge = nn.Conv1d(feature_dim, feature_dim, kernel_size=1)
```

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/29886/commits/601da1cf0756cef7339d4dd33faeefbfcd073ac9)